src/tasks/tasks.py

```diff
@@ -2224,3 +2225,152 @@ def get_metrics(self, reset=False):
         f1 = (f11 + f12 + f13) / 3
         return {'accuracy': acc, 'f1': f1, 'precision': pcs, 'recall': rcl}
 
+class MultipleChoiceTask(Task):
+    ''' Generic task class for a multiple choice
+    where each example consists of a question and
+    a (possibly variable) number of possible answers'''
+    pass
+
+
+@register_task('copa', rel_path='COPA/')
+class COPATask(MultipleChoiceTask):
+    ''' Task class for Choice of Plausible Alternatives Task.  '''
+
+    def __init__(self, path, max_seq_len, name, **kw):
+        ''' '''
+        super().__init__(name, **kw)
+        self.load_data(path, max_seq_len)
+        self.sentences = self.train_data_text[0] + self.val_data_text[0] + \
+                [choice for choices in self.train_data_text[1] for choice in choices] + \
+                [choice for choices in self.val_data_text[1] for choice in choices]
+        self.scorer1 = CategoricalAccuracy()
+        self.scorers = [self.scorer1]
+        self.val_metric = "%s_accuracy" % name
+        self.val_metric_decreases = False
+        self.n_choices = 2
+
+    def load_data(self, path, max_seq_len):
+        ''' Process the dataset located at path.  '''
+
+        def _load_split(data_file):
+            contexts, questions, choicess, targs = [], [], [], []
+            data = [json.loads(l) for l in open(data_file, encoding="utf-8")]
+            for example in data:
+                context = example["premise"]
+                choice1 = example["choice1"]
+                choice2 = example["choice2"]
+                question = example["question"]
+                question = "What was the cause of this ?" if question == "cause" else "What happened as a result ?"
```

Comment:
Is the space before '?' intentional? (i.e., is this valid/expected with all the tokenizers we might use?)